### DISCUSSION TITLE
Is there any time limit to finish a query?

### DISCUSSION THREAD

**Discussion by Etkiny** [Budibase/budibase#5139] [Help] [answered]
Hi guys,

In my project, I have a sql query which takes 15-20 seconds to execute. I need to use it as a query in Budibase. However; when I try to run the query, I get an error saying 'Error previewing the query'. What does it mean? I am sure my sql query works fine on my db console.  So, how can I execute it on Budibase.?

If it is possible with REST api, how can I put the result  into a table?

Thank you!

**Comment by mike12345567** [comment, marked as answer]
Hi @Etkiny - there is a time limit on queries, currently it is 10 seconds. If you are running in self host, you can set this limit to any number of milliseconds, through the `QUERY_THREAD_TIMEOUT` environment variable, you can set this directly on the app service in your docker compose file, e.g.
```
environment:
  ...
  QUERY_THREAD_TIMEOUT: 30000
```

If running in the Cloud, this is currently a limitation of the free tier.
